index.js

```diff
@@ -1,5 +1,85 @@
-console.log("pipocas")
+class Jogador {
+    constructor (nome,simbolo) {
+        this.nome = nome
+        this.simbolo = simbolo
+    }
+    compara(jogador) {
+        return this.nome === jogador.nome && this.simbolo === jogador.simbolo
+    }
+}
 
-function teste(context){
-    console.log(context)
-}
\ No newline at end of file
+class Tabuleiro {
+    constructor (tipotab) {
+        this.tipotab = tipotab
+        this.jogadas = new Array(tipotab*tipotab)
+        this.posvitoria = init_posvitoria(tipotab)
+    }
+}
+const tabuleiro = new Tabuleiro(3)
+const jogador1 = new Jogador("A","X")
+const jogador2 = new Jogador("B","O")
+let jogadoractual = jogador1
+
+function jogada(div){
+    if (div.innerText === '' ){
+
+    div.innerText = jogadoractual.simbolo
+    if (jogadoractual.compara(jogador1)) {
+        jogadoractual = jogador2
+    }
+    else{
+        jogadoractual = jogador1
+    }
+}
+}
+
+// [[0,1,2],[3,4,5],[6,7,8]] - check
+// [[0,3,6],[1,4,7],[2,5,8]]
+// [[0,4,8],[2,4,6]]
+
+function init_posvitoria(tipotab){
+
+let linha = []
+let total = []
+let diferenca;
+
```

Comment:
Esta é a melhor opção
diff = 0
for (let i = 0; i < 3; i++) {
    console.log('i',i)
    diff = 3 * i
    for (let j = 0; j < 3; j++) {
        console.log('j',j, 'diff', diff + j)
    }
}

Reply:
O 3 é o tipo tab